### PR TITLE
Change 'Update password' conditions to remove log file flooding

### PR DESCRIPTION
**Pull request by ledgr**
Avoid flooding log file with:

```
2016-08-26 08:30:20 2093 [Warning] Access denied for user 'root'@'localhost' (using password: NO)
2016-08-26 08:41:27 2093 [Warning] Access denied for user 'root'@'localhost' (using password: NO)
2016-08-26 08:54:25 2093 [Warning] Access denied for user 'root'@'localhost' (using password: NO)
```

### CODE REVIEW

**Comment by ledgr** [comment]
@hostinger/devops

File: recipes/configure_server.rb [hunk 1]
```diff
@@ -136,12 +136,12 @@
...
 
 # now let's set the root password only if this is the initial install
-unless node["percona"]["skip_passwords"]
+unless node["percona"]["skip_passwords"] || passwords.root_password.to_s.empty?
   root_pw = passwords.root_password
 
   execute "Update MySQL root password" do  # ~FC009 - `sensitive`
     command "mysqladmin --user=root --password='' password '#{root_pw}'"
-    only_if "mysqladmin --user=root --password='' version"
+    not_if "mysqladmin --user=root --password='#{root_pw}' version"
```
Comment (ton31337) [inline recipes/configure_server.rb:144]: what if I change the password?

Comment (ledgr) [reply recipes/configure_server.rb:144]: Well it didn't work before too. The comment above `# now let's set the root password only if this is the initial install`

Comment (ton31337) [reply recipes/configure_server.rb:144]: I think it would be better to fix this and not workaround

Comment (ton31337) [reply recipes/configure_server.rb:144]: Need to check if my.cnf is generated first and remove some parts at all

Comment (ton31337) [reply recipes/configure_server.rb:144]: Or maybe it's changed in upstream? Or not?

Comment (ton31337) [reply recipes/configure_server.rb:144]: By the way, create pull request to upstream too

File: recipes/configure_server.rb [hunk 2]
```diff
@@ -136,12 +136,12 @@
 end
 
 # now let's set the root password only if this is the initial install
-unless node["percona"]["skip_passwords"]
+unless node["percona"]["skip_passwords"] || passwords.root_password.to_s.empty?
```
Comment (ton31337) [inline recipes/configure_server.rb:139]: why second condition is needed?

File: recipes/configure_server.rb (hunk 1 above)
Comment (ton31337) [reply recipes/configure_server.rb:144]: What about removing '--password at all' for command and not_if?

**Comment by ledgr** [comment]
Setting up initial password and changing existing password is two different problems. To change a password you need to know both - old password and new password. So that adds one additional attribute and changes all the logic. I may do it with another patch and one of them will find a way to production.

`passwords.root_password.to_s.empty?` solves two problems. First it skips setting up password if it's empty (it's empty by default anyway) and second it prevents setting up a password if root_pw is `nil?` which would otherwise provide an error.

`not_if "mysqladmin --user=root --password='#{root_pw}' version"` avoids setting up a password if it's the same password that is already set. Otherwise password would be updated on every chef run even if this is not a new password.

**Comment by ton31337** [comment]
but have you tried this case?

```
command "mysqladmin password '#{root_pw}'"
not_if "mysqladmin version"
```

**Comment by ledgr** [comment]
It won't work, because it you specify a password it will end up in `.my.cnf`, then

```
command "mysqladmin password '#{root_pw}'"
not_if "mysqladmin version"
```

Will use password from `.my.cnf` and `"mysqladmin version"` will fail (password is empty, but you are trying to use one from `.my.cnf`)

Also if initial password is not set `not_if "mysqladmin version"` will skip everytime, because it will always work. And if you change it to `only_if` you are introducing other problems...

**Comment by ton31337** [comment]
- When you install new mysql server, password is empty, thus this will work:

```
command "mysqladmin password '#{root_pw}'"
not_if "mysqladmin version"
```

It will update password with new password, check should work as well.
- When password is already changed, `command "mysqladmin password '#{root_pw}'"` should work because it will use credentials from `my.cnf`, thus even check should work as well :/ I'm not right?

**Comment by ledgr** [comment]
1) Won't work because condition is `not_if` and with empty password it will always succeed.
2) Would work if first condition would work.

Changing to `only_if` would work, but that's why it was before this patch and it flooded logs with that error.

**Comment by ton31337** [comment]
What about removing that check at all? or change it to something else?

**Pull request closed without merging**